app/test/square_size_test.go

```diff
@@ -0,0 +1,262 @@
+// max bytes parameters. It assumes that the governance params have been set to
+// allow for fast acceptance of proposals, and will fail the test if the
+// parameters are not set as expected.
+func (s *IntegrationTest) setBlockSizeParams(t *testing.T, squareSize uint64, maxBytes int64) {
+	account := "validator"
+
+	bparams := &abci.BlockParams{
+		MaxBytes: maxBytes,
+		MaxGas:   -1,
+	}
+
+	// create and submit a new param change proposal for both params
+	change1 := proposal.NewParamChange(
+		blobtypes.ModuleName,
+		string(blobtypes.KeyGovMaxSquareSize),
+		fmt.Sprintf("\"%d\"", squareSize),
+	)
+	change2 := proposal.NewParamChange(
+		baseapp.Paramspace,
+		string(baseapp.ParamStoreKeyBlockParams),
+		string(s.cctx.Codec.MustMarshalJSON(bparams)),
+	)
+	content := proposal.NewParameterChangeProposal(
+		"title",
+		"description",
+		[]proposal.ParamChange{change1, change2},
+	)
+	addr := getAddress(account, s.cctx.Keyring)
+
+	msg, err := oldgov.NewMsgSubmitProposal(
+		content,
+		sdk.NewCoins(
+			sdk.NewCoin(app.BondDenom, sdk.NewInt(1000000000))),
+		addr,
+	)
+	require.NoError(t, err)
+
+	res, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, account, msg)
+	require.NoError(t, err)
+	require.NoError(t, s.cctx.WaitForNextBlock())
+	resp, err := testnode.QueryTx(s.cctx.Context, res.TxHash, false)
+	require.NoError(t, err)
+	require.Equal(t, abci.CodeTypeOK, resp.TxResult.Code)
+
+	// query the proposal to get the id
+	gqc := v1.NewQueryClient(s.cctx.GRPCClient)
+	gresp, err := gqc.Proposals(s.cctx.GoContext(), &v1.QueryProposalsRequest{ProposalStatus: v1.ProposalStatus_PROPOSAL_STATUS_VOTING_PERIOD})
+	require.NoError(t, err)
+	require.Len(t, gresp.Proposals, 1)
+
+	// create and submit a new vote
+	vote := v1.NewMsgVote(getAddress(account, s.cctx.Keyring), gresp.Proposals[0].Id, v1.VoteOption_VOTE_OPTION_YES, "")
+	res, err = testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, account, vote)
+	require.NoError(t, err)
+	require.NoError(t, s.cctx.WaitForNextBlock())
+
+	resp, err = testnode.QueryTx(s.cctx.Context, res.TxHash, false)
+	require.NoError(t, err)
+	require.Equal(t, abci.CodeTypeOK, resp.TxResult.Code)
+
+	// wait for the voting period to complete
+	time.Sleep(time.Second * 3)
+
+	// check that the parameters got updated as expected
+	bqc := blobtypes.NewQueryClient(s.cctx.GRPCClient)
+	presp, err := bqc.Params(s.cctx.GoContext(), &blobtypes.QueryParamsRequest{})
+	require.NoError(t, err)
+	require.Equal(t, squareSize, presp.Params.GovMaxSquareSize)
+
+	// unfortunately the rpc connection is very flakey with super fast block
+	// times, so we have to retry many times.
+	var newMaxBytes int64
+	for i := 0; i < 20; i++ {
+		cpresp, err := s.cctx.Client.ConsensusParams(s.cctx.GoContext(), nil)
+		if err != nil || cpresp == nil {
+			continue
+		}
+		newMaxBytes = cpresp.ConsensusParams.Block.MaxBytes
+	}
+	require.Equal(t, maxBytes, newMaxBytes)
```

Comment:
this test currently assumes that the proposed values pass, and check them here. in the future, we might want to get higher coverage by also checking that we can't go above the hardcoded values

Reply:
do we have another test that is checking that the values pass then? or put another way, we are always throttling to the lowest value?

if not that feels like a very baseline required test :-)